Backend/auction-app/src/main/java/com/example/auctionapp/repository/AbstractJpaDao.java

```diff
@@ -0,0 +1,44 @@
+package com.example.auctionapp.repository;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import java.util.List;
+
+public abstract class AbstractJpaDao<T> {
```

Comment:
We created `IBaseRepository` so that we can have an `AbstractBaseRepository`. This class should be named as such.

Reply:
`public abstract class AbstractJpaDao<T> ` -> `public abstract class BaseRepository<T> implements implements IBaseRepository<T>`